internal/test/wmcb/pkginfo.go

```diff
@@ -0,0 +1,144 @@
+package wmcb
+
+import (
+	"bufio"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+// pkgName is the user defined name of the package
+type pkgName string
+
+//cniPlugins contains information about the CNI plugin package
+type cniPlugins struct {
+	pkgInfo
+}
+
+//hybridOverlay contains information about the hybrid overlay binary
+type hybridOverlay struct {
+	pkgInfo
+}
+
+// PkgInfo is an interface to populate pkgInfo structs
+type PkgInfo interface {
+	//getPkgName returns the user defined package name
+	getPkgName() pkgName
+	//getPkgUrl returns the url of the package
+	getPkgUrl() string
+	// getSHAFileContent returns checksum file content
+	getSHAFileContent() (string, error)
+	//getSHAValue returns SHA value for the package
+	getSHAValue() (string, error)
+	// getShaType returns SHA type (e.g sha256)
+	getShaType() string
+}
+
+// pkgInfo encapsulates information about a package
+type pkgInfo struct {
+	// name of the package
+	name pkgName
+	// url is the URL of the package
+	url string
+	// sha is the SHA hash of the package
+	sha string
+	// shaType is the type of SHA used, example: 256 or 512
+	shaType string
+}
+
+// pkgInfoFactory returns PkgInfo specific to the package name
+func pkgInfoFactory(name pkgName, shaType string, baseUrl string, version string) (PkgInfo, error) {
+	switch name {
+	case cniPluginPkgName:
+		return newCniPluginPkg(name, shaType, baseUrl, version)
+	case hybridOverlayPkgName:
+		return newHybridOverlayPkg(name, shaType)
+	default:
+		return nil, fmt.Errorf("invalid Package name")
+	}
+}
+
+// newCniPluginPkg returns cniPlugins implementation of PkgInfo interface
+func newCniPluginPkg(name pkgName, shaType string, baseUrl string, version string) (PkgInfo, error) {
+	if version == "" {
+		return nil, fmt.Errorf("latest cni plugins version not specified")
+	}
+	if baseUrl == "" {
+		return nil, fmt.Errorf("base url for cni plugins not specified")
+	}
+	return &cniPlugins{pkgInfo{
+		name:    name,
+		url:     baseUrl + version + "/cni-plugins-windows-amd64-" + version + ".tgz",
+		shaType: shaType}}, nil
+}
+
+// newHybridOverlayPkg returns hybridOverlay implementation of PkgInfo interface
+func newHybridOverlayPkg(name pkgName, shaType string) (PkgInfo, error) {
+	hybridOverlayUrl, err := framework.GetReleaseArtifactURL(hybridOverlayName)
+	if err != nil {
+		return nil, err
+	}
+	return &hybridOverlay{pkgInfo{
+		name:    name,
+		shaType: shaType,
+		url:     hybridOverlayUrl}}, nil
+}
+
+// getSHAFileContent returns the contents of the SHA file for the given url of the package.
+func (p *pkgInfo) getSHAFileContent() (string, error) {
+	pkgChecksumFileURL := p.url + "." + p.shaType
+	response, err := http.Get(pkgChecksumFileURL)
+	if err != nil {
+		return "", fmt.Errorf("could not fetch content from checksum file: %s", err)
+	}
+	defer response.Body.Close()
+
+	var checksumFileContent string
+	// Fetching checksum file content from the GET Response
```

Comment:
Please address https://github.com/openshift/windows-machine-config-bootstrapper/pull/170#discussion_r398567428

Reply:
Addressed the comment in getSHAValue() function, as content format is different for every package https://github.com/openshift/windows-machine-config-bootstrapper/pull/170/files#diff-6658479ed9056a08f91d4cf1b23fda4bR137  (Line137)